src/modules/land_detector/LandDetector.cpp

```diff
@@ -178,4 +196,51 @@ void LandDetector::Run()
 	}
 }
 
+void LandDetector::UpdateVehicleAtRest()
+{
+	if (_sensor_selection_sub.updated()) {
+		sensor_selection_s sensor_selection{};
+		_sensor_selection_sub.copy(&sensor_selection);
+
+		if (sensor_selection.gyro_device_id != _device_id_gyro) {
+
+			bool gyro_status_found = false;
+
+			// find corresponding vehicle_imu_status instance
+			for (uint8_t imu_instance = 0; imu_instance < 4; imu_instance++) {
+				uORB::Subscription imu_status_sub{ORB_ID(vehicle_imu_status), imu_instance};
+
+				vehicle_imu_status_s imu_status{};
+				imu_status_sub.copy(&imu_status);
+
+				if ((imu_status.gyro_device_id != 0) && (imu_status.gyro_device_id == sensor_selection.gyro_device_id)) {
+					_vehicle_imu_status_sub.ChangeInstance(imu_instance);
+					_device_id_gyro = sensor_selection.gyro_device_id;
+					gyro_status_found = true;
+					break;
+				}
+			}
+
+			if (!gyro_status_found) {
+				PX4_WARN("IMU status not found for gyro %" PRId32, sensor_selection.gyro_device_id);
+			}
+		}
+	}
+
+	vehicle_imu_status_s imu_status;
+
+	if (_vehicle_imu_status_sub.update(&imu_status)) {
+		static constexpr float GYRO_VIBE_METRIC_MAX = 0.02f; // gyro_vibration_metric * 4.0e4f > is_moving_scaler) TODO: review
+		static constexpr float ACCEL_VIBE_METRIC_MAX = 1.2f; // accel_vibration_metric * dt * 2.1e2f > is_moving_scaler
```

Comment:
wasn't the dt = 0.05? So 2.1e-2 / 0.05 =  0.42

Reply:
5 milliseconds = 0.005 (although I had actually used the older value of 4 milliseconds).

Reply:
I had to increase the gyro threshold quite a bit before it would stop triggering in SITL.